Finished translation of "JavaScript in JSX with Curly Braces" page (#487)
reactjs/uk.react.dev: src/content/learn/javascript-in-jsx-with-curly-braces.md

---
title: JavaScript у JSX з фігурними дужками
---

<Intro>

JSX дозволяє вам писати HTML-подібну розмітку всередині файлу JavaScript, тримаючи логіку відображення та вміст в одному місці. Іноді вам потрібно додати трохи JavaScript логіки або звернутися до динамічних властивостей всередині цієї розмітки. У цій ситуації ви можете використовувати фігурні дужки у вашому JSX, щоб взаємодіяти з JavaScript.

</Intro>

<YouWillLearn>

- Як передати рядки з лапками
- Як звернутися до змінної JavaScript всередині JSX за допомогою фігурних дужок
- Як викликати функцію JavaScript всередині JSX за допомогою фігурних дужок
- Як використовувати об'єкт JavaScript всередині JSX за допомогою фігурних дужок

</YouWillLearn>

> ## Передача рядків з лапками {/*passing-strings-with-quotes*/}

Reviewer: ```suggestion
## Передача стрічок з лапками {/*passing-strings-with-quotes*/}
```

> Коли ви хочете передати рядковий атрибут в JSX, ви поміщаєте його в одинарні або подвійні лапки:

Reviewer: ```suggestion
Коли ви хочете передати атрибут-стрічку в JSX, ви поміщаєте його в одинарні або подвійні лапки:
```

function formatDate(date) {
  return new Intl.DateTimeFormat(

>     'uk-UA',

Author: In my opinion, this code should result in "Список справ на вівторок", not "Список справ на Tuesday"

</Sandpack>

У цьому прикладі JavaScript об'єкт `person` містить рядок `name` та об'єкт `theme`:

```js
const person = {
  name: 'Грегоріо І. Зара (Gregorio Y. Zara)',
  theme: {
    backgroundColor: 'black',
    color: 'pink'
  }
};
```

Компонент може використовувати ці значення з `person` так:

```js
<div style={person.theme}>
  <h1>{person.name}'s Справи</h1>
```

> JSX є дуже мінімальною мовою шаблонізації, оскільки вона дозволяє вам організовувати дані та логіку за допомогою JavaScript.

Reviewer: ```suggestion
JSX є дуже мінімальною, як мова шаблонізації, оскільки вона дозволяє вам організовувати дані та логіку за допомогою JavaScript.
```

<Recap>

Тепер ви майже все знаєте про JSX:

* Атрибути JSX всередині лапок передаються як рядки.
* Фігурні дужки дозволяють вам використовувати логіку та змінні JavaScript у вашій розмітці.
* Вони працюють всередині вмісту тегу JSX або безпосередньо після `=` в атрибутах.
* `{{` та `}}` - це не спеціальний синтаксис: це об'єкт JavaScript, який знаходиться всередині фігурних дужок JSX.

</Recap>

<Challenges>

#### Виправте помилку {/*fix-the-mistake*/}

> Цей код видає помилку, кажучи: `Об'єкти не є допустимими як дитина React`:

Reviewer: ```suggestion
Цей код видає помилку `Objects are not valid as a React child`:
```
Я б не став перекладати вміст error messages

</Sandpack>

Ви можете знайти проблему?

> <Hint>Подивіться, що знаходиться всередині фігурних дужок. Чи ми кладемо туди правильні дані?</Hint>

Reviewer: ```suggestion
<Hint>Подивіться, що знаходиться всередині фігурних дужок. Чи кладемо ми правильні дані туди?</Hint>
```
На мій погляд, так буде більш милозвучно.

</Solution>

#### Напишіть вираз всередині фігурних дужок JSX {/*write-an-expression-inside-jsx-curly-braces*/}

> У наведеному нижче об'єкті повний URL зображення розбитий на чотири частини: базовий URL, `imageId`, `imageSize`, та розширення файлу.

Reviewer: ```suggestion
У наведеному нижче об'єкті повний URL зображення розбитий на чотири частини: основний URL, `imageId`, `imageSize`, та розширення файлу.
```
